### ISSUE TITLE
Change loading of connections to use CSA

### ISSUE THREAD

**Issue by rowleya** [SpiNNakerManchester/sPyNNaker#81]
Connections are currently expanded on the host.  We would like to change this to use Connection Set Algerbra to make it possible to expand up the connections on the machine.  This would also have the advantage that new connector types can just be described using CSA, avoiding the need to do more work on the machine.

**Comment by andrewgait** [comment]
Certain connections are now expanded on the machine (not using CSA).  This is probably still an open issue for now in case we decide to implement CSA on-machine rather than on-host as it currently is.

**Comment by andrewgait** [comment]
This is definitely not something to be considered for the upcoming release.

**Comment by rowleya** [comment]
Yes - this is sort-of done in that connections get expanded on the machine, but CSA on the machine would certainly allow more powerful generation of connections

**Comment by Christian-B** [comment]
all connections we think are expandable of machine now are.
